shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Core.hs

```diff
@@ -40,6 +48,13 @@ type family TxBody era :: Type
 -- | Scripts which may lock transaction outputs in this era
 type family Script era :: Type
 
+-- | Update system environment
+type family UPEnv era :: Type
+
+-- | STS's that can make use of update rules.
+class EmbedsUpdateLogic t era where
```

Comment:
This is basically a specific `HasField` constraint. Since we're using `HasField` in other places, it would be nicer to use it here as well. Note that one can write manual instances, so it works even if e.g. the fields don't have names.